dump-event-schema-compiler-plugin/src/main/scala/org/virtuslab/akkasaferserializer/DumpEventSchemaOptions.scala

```diff
@@ -0,0 +1,3 @@
+package org.virtuslab.akkasaferserializer
+
+class DumpEventSchemaOptions(var outputDir: String, var verbose: Boolean)
```

Comment:
Hmmm only noticed that now... why is it a regular class with mutable state (`var`) rather than a case class? 🤔

Reply:
The problem is that scala compiler is very stateful.
Components must be declared here:
https://github.com/VirtusLab/akka-safer-serializer/blob/84f35b34be945c60525475c1840c8026397c9eb8/checker-plugin/src/main/scala/org/virtuslab/akkasaferserializer/AkkaSerializabilityCheckerPlugin.scala#L13
But the configuration is received by this method:
https://github.com/VirtusLab/akka-safer-serializer/blob/84f35b34be945c60525475c1840c8026397c9eb8/checker-plugin/src/main/scala/org/virtuslab/akkasaferserializer/AkkaSerializabilityCheckerPlugin.scala#L15
Do you know any way to make it more fp-like?

Reply:
Okay ☹️ pls document the purpose of using vars here maybe then since it's a very uncommon view in Scala :/